src/main/java/org/folio/service/mapping/RuleFactory.java

```diff
@@ -61,46 +68,75 @@
+  private static final String SET_METADATA_CREATED_DATE_FIELD_ID = "metadata.createddate";
   private static final String SET_LOCATION_FUNCTION = "set_location";
   private static final String MATERIAL_TYPE_FIELD_ID = "materialTypeId";
   private static final String INSTANCE_TYPE_FIELD_ID = "instanceTypeId";
   private static final String SET_MATERIAL_TYPE_FUNCTION = "set_material_type";
   private static final String SET_INSTANCE_TYPE_ID_FUNCTION = "set_instance_type_id";
+  private static final String SET_METADATA_UPDATED_DATE_FUNCTION = "set_transaction_datetime";
+  private static final String SET_METADATA_CREATED_DATE_FUNCTION = "set_fixed_length_data_elements";
 
-  private static final Map<String, String> translationFunctions = ImmutableMap.of(
-    PERMANENT_LOCATION_FIELD_ID, SET_LOCATION_FUNCTION,
-    TEMPORARY_LOCATION_FIELD_ID, SET_LOCATION_FUNCTION,
-    EFFECTIVE_LOCATION_FIELD_ID, SET_LOCATION_FUNCTION,
-    MATERIAL_TYPE_FIELD_ID, SET_MATERIAL_TYPE_FUNCTION,
-    INSTANCE_TYPE_FIELD_ID, SET_INSTANCE_TYPE_ID_FUNCTION
-  );
+
+  private static final Map<String, String> translationFunctions = ImmutableMap.<String, String>builder()
+    .put(PERMANENT_LOCATION_FIELD_ID, SET_LOCATION_FUNCTION)
+    .put(TEMPORARY_LOCATION_FIELD_ID, SET_LOCATION_FUNCTION)
+    .put(EFFECTIVE_LOCATION_FIELD_ID, SET_LOCATION_FUNCTION)
+    .put(MATERIAL_TYPE_FIELD_ID, SET_MATERIAL_TYPE_FUNCTION)
+    .put(INSTANCE_TYPE_FIELD_ID, SET_INSTANCE_TYPE_ID_FUNCTION)
+    .put(SET_METADATA_UPDATED_DATE_FIELD_ID, SET_METADATA_UPDATED_DATE_FUNCTION)
+    .put(SET_METADATA_CREATED_DATE_FIELD_ID, SET_METADATA_CREATED_DATE_FUNCTION)
+    .build();
 
   private List<Rule> defaultRules;
 
   public List<Rule> create(MappingProfile mappingProfile) {
+    return create(mappingProfile, getDefaultRulesFromFile());
+  }
+
+  public List<Rule> create(MappingProfile mappingProfile, List<Rule> defaultRules) {
     if (mappingProfile == null || isEmpty(mappingProfile.getTransformations())) {
       LOGGER.info("No Mapping rules specified, using default mapping rules");
-      return getDefaultRules();
+      return defaultRules;
     }
-    List<Rule> rules = Lists.newArrayList(getDefaultRules());
-    rules.addAll(buildByTransformations(mappingProfile.getTransformations()));
-    return rules;
+    return new ArrayList<>(createByTransformations(mappingProfile.getTransformations(), defaultRules));
   }
 
-  protected List<Rule> getDefaultRules() {
-    if (Objects.nonNull(this.defaultRules)) {
-      return this.defaultRules;
+  public Set<Rule> createByTransformations(List<Transformations> mappingTransformations, List<Rule> defaultRules) {
+    Set<Rule> rules = new LinkedHashSet<>();
+    String temporaryLocationTransformation = getTemporaryLocationTransformation(mappingTransformations);
+    for (Transformations mappingTransformation : mappingTransformations) {
+      if (isTransformationValid(mappingTransformation) && isNotBlank(mappingTransformation.getTransformation())
```

Comment:
CAn you please move this long condition to a predicate or something, it's difficult to read what it's doing

Reply:
done